packages/frontend-api/src/Model/Resolver/Products/DataMapper/ProductArrayFieldMapper.php

```diff
@@ -0,0 +1,236 @@
+     */
+    protected ParameterWithValuesFactory $parameterWithValuesFactory;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryFacade $categoryFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade $flagFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\BrandFacade $brandFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider $productElasticsearchProvider
+     * @param \Shopsys\FrontendApiBundle\Model\Parameter\ParameterWithValuesFactory $parameterWithValuesFactory
+     */
+    public function __construct(
+        CategoryFacade $categoryFacade,
+        FlagFacade $flagFacade,
+        BrandFacade $brandFacade,
+        ProductElasticsearchProvider $productElasticsearchProvider,
+        ParameterWithValuesFactory $parameterWithValuesFactory
+    ) {
+        $this->categoryFacade = $categoryFacade;
+        $this->flagFacade = $flagFacade;
+        $this->brandFacade = $brandFacade;
+        $this->productElasticsearchProvider = $productElasticsearchProvider;
+        $this->parameterWithValuesFactory = $parameterWithValuesFactory;
+    }
+
+    /**
+     * @param array $data
+     * @return string|null
+     */
+    public function getShortDescription(array $data): ?string
+    {
+        return $data['short_description'];
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public function getLink(array $data): string
+    {
+        return $data['detail_url'];
+    }
+
+    /**
+     * @param array $data
+     * @return \Shopsys\FrameworkBundle\Model\Category\Category[]
+     */
+    public function getCategories(array $data): array
+    {
+        return $this->categoryFacade->getByIds($data['categories']);
+    }
+
+    /**
+     * @param array $data
+     * @return \Shopsys\FrameworkBundle\Model\Product\Flag\Flag[]
+     */
+    public function getFlags(array $data): array
+    {
+        return $this->flagFacade->getByIds($data['flags']);
+    }
+
+    /**
+     * @param array $data
+     * @return string[]
+     */
+    public function getAvailability(array $data): array
+    {
+        return ['name' => $data['availability']];
+    }
+
+    /**
+     * @param array $data
+     * @return string[]
+     */
+    public function getUnit(array $data): array
+    {
+        return ['name' => $data['unit']];
+    }
+
+    /**
+     * @param array $data
+     * @return int|null
+     */
+    public function getStockQuantity(array $data): ?int
+    {
+        return $data['stock_quantity'];
+    }
+
+    /**
+     * @param array $data
+     * @return bool
+     */
+    public function getIsUsingStock(array $data): bool
+    {
+        return $data['is_using_stock'];
+    }
+
+    /**
+     * @param array $data
+     * @return \Shopsys\FrameworkBundle\Model\Product\Brand\Brand|null
+     */
+    public function getBrand(array $data): ?Brand
+    {
+        if ((int)$data['brand'] > 0) {
```

Comment:
`(int)[2, 3]` will return `1`, so it passes the condition too. I am not sure if it is OK or not 🤔. Probably it should be handled from type definitions?

Reply:
Well, the behavior is not changed, the code was just moved.
Not changed as each product can have only one brand (see ProductExportRepository)

Reply:
Understand. The condition just seems weird to me. It could be `$data['brand'] !== ''` which could make it more obvious about incoming data. I will leave it up to you.